src/main/resources/hudson/tasks/junit/History/index.jelly

```diff
@@ -20,88 +20,144 @@ AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
   <bs:layout title="${%title(it.testObject.displayName)}">
+    <j:set var="count" value="${it.asInt(request.getParameter('count'),100)}"/>
     <j:set var="start" value="${it.asInt(request.getParameter('start'),0)}"/>
-    <j:set var="end" value="${it.asInt(request.getParameter('end'),start+24)}"/>
+    <j:set var="end" value="${it.asInt(request.getParameter('end'),start+count-1)}"/>
+    <j:set var="historySummary" value="${it.retrieveHistorySummary(start, end)}"/>
+    <j:set var="trendChartJson" value="${historySummary.trendChartJson}"/>
+    <j:set var="trendChartJsonStr" value="${historySummary.trendChartJsonStr}"/>
+    <j:set var="buildMap" value="${historySummary.buildMapJson}"/>
     <st:adjunct includes="io.jenkins.plugins.echarts"/>
     <st:adjunct includes="io.jenkins.plugins.bootstrap5"/>
     <st:adjunct includes="io.jenkins.plugins.echarts-trend-default-setup"/>
     <link rel="stylesheet" href="${resURL}/plugin/junit/history/history.css"/>
 
     <st:once>
       <c:chart-setup id="test-history"/>
-      <script>var view =
-        <st:bind value="${it}"/>
+      <script>
+      var view = <st:bind value="${it}"/>
+      var start = ${start}
+      var end = ${end}
+      var trendChartJson = ${trendChartJson}
+      var trendChartJsonStr = ${trendChartJsonStr}
+      var rootUrl = "${app.rootUrl}"
       </script>
       <script type="text/javascript" src="${resURL}/plugin/junit/history/history.js"/>
     </st:once>
 
     <st:include from="${it.testObject}" it="${it.testObject}" page="sidepanel.jelly"/>
     <l:main-panel>
-      <j:choose>
-        <j:when test="${it.historyAvailable()}">
-          <bs:card title="${%History}" fontAwesomeIcon="chart-line">
-            <div id="trend-carousel" class="carousel slide carousel-dark" data-bs-interval="false">
-              <div class="carousel-inner">
-                <div class="carousel-item">
-                  <div id="test-duration-trend-chart" class="graph-cursor-pointer card-chart-carousel"/>
-                </div>
-                <div class="carousel-item active">
-                  <div id="test-result-trend-chart" class="graph-cursor-pointer card-chart-carousel"/>
-                </div>
-              </div>
-              <button class="carousel-control-prev" type="button" data-bs-target="#trend-carousel" data-bs-slide="prev">
-                <span class="carousel-control-prev-icon" aria-hidden="true"/>
-                <span class="visually-hidden">Previous</span>
-              </button>
-              <button class="carousel-control-next" type="button" data-bs-target="#trend-carousel" data-bs-slide="next">
-                <span class="carousel-control-next-icon" aria-hidden="true"/>
-                <span class="visually-hidden">Next</span>
-              </button>
-            </div>
-          </bs:card>
+      <div style="margin-bottom: 0.5em; margin-top: -1.75em; height: 1.5em;">
```

Comment:
avoid inline styles and use from the design library instead for standard spacing: https://weekly.ci.jenkins.io/design-library/Spacing/